### DISCUSSION TITLE
8.10 背包问题（五）: 背包问题变种

### DISCUSSION THREAD

**Discussion by giscus** [itcharge/AlgoNote#107] [Comments]
# 8.10 背包问题（五）: 背包问题变种

[https://algo.itcharge.cn/08_dynamic_programming/08_10_knapsack_problem_05/](https://algo.itcharge.cn/08_dynamic_programming/08_10_knapsack_problem_05/)

**Comment by sadcoldpig** [comment]
8.2 求方案总数 ：如果至多W的话，应该将dp全部初始化为1才对吧

**Comment by gxpde** [comment]
求方案总数的0-1 背包问题的二维状态转移方程感觉不对，应该是
dp[i][w]=dp[i−1][w]+dp[i-1][w−weight[i−1]]
